syntax = "proto3";

import "google/protobuf/empty.proto";

service Users {
    rpc CreateUser(CreateUserRequest) returns (CreateUserResponse);
    rpc ReadUser(ReadUserRequest) returns (ReadUserResponse);
    rpc UpdateUser(UpdateUserRequest) returns (UpdateUserResponse);
    rpc DeleteUser(DeleteUserRequest) returns (google.protobuf.Empty);
}

message User {
    string id = 1;
    string email = 2;
}

message CreateUserRequest {
    string email = 1;
    optional string password = 3;
}

message CreateUserResponse {
    User user = 1;
}

message ReadUserRequest {
    string email = 1;
}

message ReadUserResponse {
    User user = 1;
}

message UpdateUserRequest {
    User updated_user = 1;
}

message UpdateUserResponse {
    User updated_user = 1;
}

message DeleteUserRequest {
    string email = 1;
}

message DeleteUserResponse {
    bool success = 1;
    string message = 2;
}
